### ISSUE TITLE
Note rendered outside document's bounds

### ISSUE THREAD

**Issue by automatedbugreportingfacility** [mozilla/pdf.js#6230]
I thought I'd post this seeing the recent round of annotation fixes. This is a regression from PR #4318, at least partial, because prior to the change, the note's color was wrong (yellow).

URL: http://www.nyc.gov/html/dcp/pdf/zone/map8c.pdf

076b3433b4de05a51716bbeccd6a839a3d0a8d6c is the first bad commit
commit 076b3433b4de05a51716bbeccd6a839a3d0a8d6c
Author: Samuel Chantaraud
Date:   Fri Mar 7 10:48:42 2014 -0400

    Improved annotations' display/behavior.

    Added an "InteractiveAnnotation" class to homogenize the annotations' structure (highlighting) and user interactions (for now, used for text and link annotations).

    Text annotations:
    The appearance (AP) has priority over the icon (Name).
    The popup extends horizontally (up to a limit) as well as vertically.
    Reduced the title's font size.
    The annotation's color (C) is used to color the popup's background.
    On top of the mouseover show/hide behavior, a click on the icon will lock the annotation open (for mobile purposes). It can be closed with another click on either the icon or the popup.

    An annotation printing is conditioned by its "print" bit
    Unsupported annotations are not displayed at all.

@timvandermeij

**Comment by automatedbugreportingfacility** [comment]
On a second thought, users can click the annotation, pinning it in the position, and use the horizontal slider to reveal the hidden part of the annotation. Adobe Acrobat Reader moves the annotation to the left when there's not enough space to render the annotation, though.

**Comment by timvandermeij** [comment]
So to be sure I understand the problem correctly, the only issue here is that the pop-up extends horizontally, causing it to be partially out of view? (The color and position of the icon indeed are the same as in Adobe Acrobat for me.)

**Comment by automatedbugreportingfacility** [comment]
Exactly. Actually, the horizontal extension is correct (as opposed to the vertical before PR #4318) -- it's just that it shouldn't extend beyond the edge of the screen, instead the position of the annotation should be adjusted to the left, such that the right edge of the annotation aligns with the edge of the screen.

**Comment by Snuffleupagus** [comment]
> I noticed that annotation pop-ups can be rendered outside of the pdf document.

Yes, and this issue is already *exactly* about that.

In the default viewer this should only be a minor inconvenience, since as mentioned in https://github.com/mozilla/pdf.js/issues/6230#issuecomment-122438789 the user should still be able to access such PopupAnnotations just fine in practice.

The reason that this issue hasn't been fixed yet, despite its age, is likely because it's a somewhat minor issue that's probably not trivial to address without a bunch of complexity (since it's easy to imagine various edge-cases).
